Riot Share Extension/ShareViewController.m

```diff
@@ -0,0 +1,195 @@
+/*
+ Copyright 2017 Aram Sargsyan
+ 
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+ 
+ http://www.apache.org/licenses/LICENSE-2.0
+ 
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import "ShareViewController.h"
+#import "SegmentedViewController.h"
+#import "RoomsListViewController.h"
+#import "FallbackViewController.h"
+
+
+@interface ShareViewController ()
+
+@property (nonatomic) NSArray <MXRoom *> *rooms;
+
+@property (weak, nonatomic) IBOutlet UIView *masterContainerView;
+@property (weak, nonatomic) IBOutlet UILabel *tittleLabel;
+@property (weak, nonatomic) IBOutlet UIView *contentView;
+
+@property (nonatomic) SegmentedViewController *segmentedViewController;
+
+
+@end
+
+
+@implementation ShareViewController
+
+#pragma mark - Lifecycle
+
+- (void)viewDidLoad
+{
+    [super viewDidLoad];
+    
+    [self prepareSession];
+    [self configureViews];
+    
+    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(onSessionSync:) name:kMXSessionDidSyncNotification object:nil];
+}
+
+- (void)viewWillDisappear:(BOOL)animated
+{
+    [super viewWillDisappear:animated];
+    
+    [[NSNotificationCenter defaultCenter] removeObserver:self];
+}
+
+#pragma mark - Private
+
+- (void)prepareSession
+{
+    // Prepare account manager
+    MXKAccountManager *accountManager = [MXKAccountManager sharedManager];
+    
+    // Use MXFileStore as MXStore to permanently store events.
+    accountManager.storeClass = [MXFileStore class];
+    
+    // Start a matrix session for each enabled accounts.
+    NSLog(@"[AppDelegate] initMatrixSessions: prepareSessionForActiveAccounts");
+    [accountManager prepareSessionForActiveAccounts];
+    
+    // Resume all existing matrix sessions
+    NSArray *mxAccounts;// = accountManager.activeAccounts;
```

Comment:
It was done for you to see the new Fallback view, will be brought back of course